packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,297 @@
+import {helpers, ShopperLogin, ShopperLoginTypes} from 'commerce-sdk-isomorphic'
+import jwtDecode from 'jwt-decode'
+import {ApiClientConfigParams} from '../hooks/types'
+import {BaseStorage, LocalStorage, CookieStorage} from './storage'
+
+type Helpers = typeof helpers
+interface AuthConfig {
+    shortCode: string
+    organizationId: string
+    clientId: string
+    siteId: string
+    redirectURI: string
+    proxy: string
+}
+
+interface JWTHeaders {
+    exp: number
+    iat: number
+}
+
+// this type is slightly different from ShopperLoginTypes.TokenResponse, reasons:
+// 1. TokenResponse is too generic (with & {[key:string]: any}), we need a more
+//    restrictive type to make sure type safe
+// 2. The refresh tokens are stored separately for guest and registered user. Instead
+//    of refresh_token, we have refresh_token_guest and refresh_token_registered
+type AuthDataKeys =
+    | 'access_token'
+    | 'customer_id'
+    | 'enc_user_id'
+    | 'expires_in'
+    | 'id_token'
+    | 'idp_access_token'
+    | 'refresh_token_guest'
+    | 'refresh_token_registered'
```

Comment:
With respect to you comment above and these 2 values, why is it that we need to have 2 distinct values? Any any given point you shouldn't have a defined guest and a define registered value, correct? So why do we need this distinction when it can be derived from the other values if the user is reg or guest?

Reply:
This is for plugin_slas integration.

To make sure PWA is compatible with plugin_slas, a requirement is that we need to separate refresh tokens and make sure the keys follow the same values `cc-nx` and `cc-nx-g`.